File: Low Level Design (LLD)/Problems/05. FoodDeliverySystem/Java/models/Order.java
package models;

import utils.OrderStatus;

import java.util.List;

public class Order {
    private String id;
    private User customer;
    private Restaurant restaurant;
    private List<FoodItem> items;
    private double totalAmount;
    private OrderStatus status;

    public Order(String id, User customer, Restaurant restaurant, List<FoodItem> items, double totalAmount, OrderStatus status) {
        this.id = id;
        this.customer = customer;
        this.restaurant = restaurant;
        this.items = items;
        this.totalAmount = totalAmount;
        this.status = status;
    }

    public String getId() {
        return id;
    }

    public User getCustomer() {
        return customer;
    }

    public Restaurant getRestaurant() {
        return restaurant;
    }

    public List<FoodItem> getItems() {
        return items;
    }

    public double getTotalAmount() {
        return totalAmount;
    }

    public OrderStatus getStatus() {
        return status;
    }

    public void setStatus(OrderStatus status) {
        this.status = status;
    }
}

File: Low Level Design (LLD)/Problems/05. FoodDeliverySystem/Cpp/models/src/Order.cpp
#include "../include/Order.h"

Order::Order(const std::string& id, const User& customer, const Restaurant& restaurant,
             const std::vector<FoodItem>& items, double totalAmount, OrderStatus status)
    : id(id), customer(customer), restaurant(restaurant),
      items(items), totalAmount(totalAmount), status(status) {}

const std::string& Order::getId() const {
    return id;
}

const User& Order::getCustomer() const {
    return customer;
}

const Restaurant& Order::getRestaurant() const {
    return restaurant;
}

const std::vector<FoodItem>& Order::getItems() const {
    return items;
}

double Order::getTotalAmount() const {
    return totalAmount;
}

OrderStatus Order::getStatus() const {
    return status;
}

void Order::setStatus(OrderStatus status) {
    this->status = status;
}
